### PR TITLE
RWA-587: Added Task db, schema and migrations

### PR DESCRIPTION
**Pull request by alexbaeza**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-587

### Change description ###

Add new postgres 11 database through terraform configuration
Added secrets to env vars in values.yaml, jenkins and bootstrap.yaml
Added dependencies and preview chart
Added initial schema 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -37,6 +39,8 @@ withPipeline(type, product, component) {
     disableLegacyDeployment()
     loadVaultSecrets(secrets)
 
+    env.POSTGRES_HOST = "wa-task-management-api-postgres-db-aat.postgres.database.azure.com"
```
Comment (raghera) [inline Jenkinsfile_CNP:42]: see my comment on https://github.com/hmcts/wa-task-management-api/pull/267/files about the name of the DB.

File: build.gradle [hunk 2]
```diff
@@ -200,9 +201,9 @@ sonarqube {
     property "sonar.pitest.mode", "reuseReport"
     property "sonar.pitest.reportsDirectory", "build/reports/pitest"
     property "sonar.exclusions", "src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/config/**," +
-                                 "src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/Application.java" +
-                                 "src/testUtils/java/uk/gov/hmcts/reform/wataskmanagementapi/**," +
-                                 "src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/clients/**"
+        "src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/Application.java" +
```
Comment (raghera) [inline build.gradle:204]: seems like different people have different format settings?

Comment (alexbaeza) [reply build.gradle:204]: looks like my groovy settings have changed from 2 spaces to 4 spaces on continuation i will change to 2

File: Jenkinsfile_CNP (hunk 1 above)
Comment (alexbaeza) [reply Jenkinsfile_CNP:42]: Thanks will apply change

**Review by raghera — COMMENTED** [review]
Looks really good in general.  Just the comment about the db/schema name.

**Review by iguannago — DISMISSED** [review]

File: src/main/resources/db/migration/V1.0.2__init_enums.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
+DROP TYPE IF EXISTS task_state_enum;
+CREATE TYPE task_state_enum as ENUM (
+    'UNCONFIGURED',
+    'PENDING_AUTO_ASSIGN',
+    'ASSIGNED',
+    'UNASSIGNED',
+    'COMPLETED',
+    'CANCELLED',
+    'TERMINATED',
+    'PENDING_RECOMMENDATION');
```
Comment (BenNzewi) [inline src/main/resources/db/migration/V1.0.2__init_enums.sql:10]: should this be PENDING_RECONFIGURATION as per confluence documentation

File: src/main/resources/db/migration/V1.0.3__init_tables.sql [hunk 4]
```diff
@@ -0,0 +1,72 @@
+DROP TABLE IF EXISTS execution_types;
+CREATE TABLE execution_types
+(
+    execution_code execution_type_enum,
+    execution_name TEXT,
+    description    TEXT,
+    PRIMARY KEY (code)
```
Comment (BenNzewi) [inline src/main/resources/db/migration/V1.0.3__init_tables.sql:7]: should the PRIMARY KEY = execution_code

File: src/main/resources/db/migration/V1.0.3__init_tables.sql [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    region_name             TEXT,
+    location                TEXT,
+    location_name           TEXT,
+    business_context        business_context_enum,
+    termination_reason      TEXT,
+    created                 TIMESTAMP default CURRENT_TIMESTAMP,
+    PRIMARY KEY (task_id),
+    CONSTRAINT fk_execution_type_code
+        FOREIGN KEY (execution_type_code)
+            REFERENCES execution_types (code)
```
Comment (BenNzewi) [inline src/main/resources/db/migration/V1.0.3__init_tables.sql:47]: FOREIGN KEY (execution_code)?

File: src/main/resources/db/migration/V1.0.3__init_tables.sql [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+);
+
+DROP TABLE IF EXISTS task_roles;
+CREATE TABLE task_roles
+(
+    task_role_id        UUID,
+    read                BOOLEAN   default false,
```
Comment (BenNzewi) [inline src/main/resources/db/migration/V1.0.3__init_tables.sql:54]: missing attribute role_name:text

File: src/main/resources/db/migration/V1.0.2__init_enums.sql (hunk 3 above)
Comment (alexbaeza) [reply src/main/resources/db/migration/V1.0.2__init_enums.sql:10]:
```suggestion
    'PENDING_RECONFIGURATION');
```

File: src/main/resources/db/migration/V1.0.3__init_tables.sql (hunk 4 above)
Comment (alexbaeza) [reply src/main/resources/db/migration/V1.0.3__init_tables.sql:7]:
```suggestion
    PRIMARY KEY (execution_code)
```

File: src/main/resources/db/migration/V1.0.3__init_tables.sql (hunk 5 above)
Comment (alexbaeza) [reply src/main/resources/db/migration/V1.0.3__init_tables.sql:47]: that one is okay the reference however should change

Comment (alexbaeza) [reply src/main/resources/db/migration/V1.0.3__init_tables.sql:47]:
```suggestion
            REFERENCES execution_types (execution_code)
```

File: src/main/resources/db/migration/V1.0.3__init_tables.sql (hunk 6 above)
Comment (alexbaeza) [reply src/main/resources/db/migration/V1.0.3__init_tables.sql:54]:
```suggestion
    role_name           TEXT,
    read                BOOLEAN   default false,
```

**Review by indvamshi — APPROVED** [review]

**Pull request merged**